transformations/multilingual_code_switch/test.json

```diff
@@ -0,0 +1,60 @@
+{
+  "type": "multilingual_code_switch",
+  "test_cases": [
+    {
+      "class": "MultilingualCodeSwitch",
+      "inputs": {
+        "sentence": "Andrew finally returned the French book to Chris that I bought last week"
+      },
+      "outputs": [
+        {
+          "sentence": "Andrew اخيرا returned the French réserver to Chris that I купила 最後 सप्ताह"

```

Comment:
@samsontmr  Hi, samson. Thanks for your insightful suggestions. We have updated it according to your advice;
(1) We renamed our method and changed the bib;
(2) We have highlighted the random translation chosen in README.md. This is a very good question. 

We found that though we cannot guarantee that this is the correct word-to-word translation in the context, we can consider it as one of the data augmented strategies for cross-lingual zero-shot tasks. Actually, we also highlighted this in our paper. 

Thanks again!

Reply:
It seems that similar implementations of code-switching perturbation have existed in NL-Augmenter.

The [multilingual_lexicon_perturbation](https://github.com/GEM-benchmark/NL-Augmenter/tree/main/transformations/multilingual_lexicon_perturbation) uses a multilingual lexicon dictionary, and randomly replaces words into the corresponding translation in the target language according to the dictionary. This is very similar to this implementation.

The [mixed_language_perturbation](https://github.com/GEM-benchmark/NL-Augmenter/tree/main/transformations/mixed_language_perturbation) is a heavy version, which uses an off-the-shelf translation model to get the code-switching candidates.

Could you please demonstrate sufficient difference from these existing transformations (especially the first one), otherwise it might seem a bit overlapped.

Reply:
@Nickeilf Hi, zhenhao. Thanks for your kind comments. This is a very good question. 

The main difference is that the mentioned transformation can only perform one-to-one mapping, which means that they only consider a pair of source and target languages at a time. By contrast, our transformation performs one-to-many mapping to construct multi-lingual code-switched data in multiple languages. 

By doing this, contextualized embeddings for all the languages can be aligned into the same space while prior work can only align representation in source and one target language for each language. Aligning multiple languages at one time is the core contribution in our work and we have highlighted it in our accepted paper[1].

Thanks for your kind suggestions again!

[1] Qin et al. CoSDA-ML: Multi-Lingual Code-Switching Data Augmentation for Zero-Shot Cross-Lingual NLP. IJCAI2020.

Reply:
Makes sense. Thanks for the explanation.